app/src/main/java/com/framgia/fsalon/screen/booking/detail/DetailFragment.java

```diff
@@ -24,8 +31,11 @@ public static DetailFragment newInstance() {
     @Override
     public void onCreate(@Nullable Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
-        mViewModel = new DetailViewModel();
-        DetailContract.Presenter presenter = new DetailPresenter(mViewModel);
+        mViewModel = new DetailViewModel(this);
+        DetailContract.Presenter presenter = new DetailPresenter(mViewModel, new UserRepository
+            (new UserRemoteDataSource(FSalonServiceClient.getInstance()), new UserLocalDataSource
```

Comment:
ERROR: 'operator new lparen' have incorrect indentation level 12, expected level should be 8.